html5ever/tests/serializer.rs

```diff
@@ -28,33 +102,65 @@ fn parse_and_serialize(input: StrTendril) -> StrTendril {
     StrTendril::try_from_byte_slice(&result).unwrap()
 }
 
-macro_rules! test {
-    ($name:ident, $input:expr, $output:expr) => {
+macro_rules! test_fn {
+    ($f:ident, $name:ident, $input:expr, $output:expr) => {
         #[test]
         fn $name() {
-            assert_eq!($output, &*parse_and_serialize($input.to_tendril()));
+            assert_eq!($output, &*$f($input.to_tendril()));
         }
     };
 
     // Shorthand for $output = $input
-    ($name:ident, $input:expr) => {
-        test!($name, $input, $input);
+    ($f:ident, $name:ident, $input:expr) => {
+        test_fn!($f, $name, $input, $input);
+    };
+}
+
+macro_rules! test {
+    ($($t:tt)*) => {
+        test_fn!(parse_and_serialize, $($t)*);
     };
 }
 
+macro_rules! test_no_parse {
+    ($($t:tt)*) => {
+        test_fn!(tokenize_and_serialize, $($t)*);
+    };
+}
+
+
+
 test!(empty, r#""#);
 test!(smoke_test, r#"<p><i>Hello</i>, World!</p>"#);
 
-test!(misnest, r#"<p><i>Hello!</p>, World!</i>"#,
-    r#"<p><i>Hello!</i></p><i>, World!</i>"#);
+test!(
```

Comment:
Ah, yeah. Thought I `git add -p`'d around them, but I guess not. Should be an easy change.